deposit/declaration.py

```diff
@@ -0,0 +1,256 @@
+import io
+import logging
+import os
+import sys
+
+from copy import copy
+from datetime import date
+
+from reportlab.lib.pagesizes import A4
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.lib.units import mm
+
+from reportlab.platypus import Paragraph
+
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+
+from reportlab.pdfgen import canvas
+
+from dissemin.settings import BASE_DIR
+
+logger = logging.getLogger('dissemin.' + __name__)
+
+# Get the current module name to load functions by name
+current_module = sys.modules[__name__]
+
+# The dict contains the available functions and an admin-friendly name
+REGISTERED_DECLARATION_FUNCTIONS = {
+    'declaration_ulb_darmstadt' : 'ULB Darmstadt',
+}
+
+
+def get_declaration_pdf(deposit_record, user):
+    """
+    This function creates for a given deposit the letter of declaration. If the does not succeed, it raises an exception.
+    :param deposit: DepositRecord containing information for declaration
+    :returns: BytesIO containing pdf or raises Exception
+    """
+    func = get_declaration_function(deposit_record.repository.letter_declaration)
+    return func(deposit_record, user)
+
+
+def get_declaration_function(func_name):
+    """
+    This function gets the function to generate a letter of declaration. If no function can be found, we raise AttributeError.
+    :param func_name: Name of the function that is requested
+    :returns: function or raises AttributeError
+    """
+    return getattr(current_module, func_name)
+
+
+def declaration_ulb_darmstadt(deposit_record, user):
+    """
+    Takes a deposit and creates authors declaration for ULB Darmstadt and returns that.
+    This function follows the corporate design of TUDA.
+    As this is my first approach with reportlab, the way placing things is not really good
+    """
+
+    pdf_buffer = io.BytesIO()
```

Comment:
This approach of generating the PDF file from scratch with `reportlab` in Python is definitely fine for me. Not sure though, but would it be possible to have instead a source PDF file with forms and just pre-fill the forms in Dissemin?

My idea behind this is that it would be easier to add new repositories since the contributor would just have to contribute a PDF file with forms, which can be easily generated from any PDF creation software.

Definitely not a blocking point for merging though.

Reply:
Yes, I have thought about that, too. I prefer this way to, but it seemed as the more difficult approach to me.

I'll do some investigation on that and open an issue.